packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+    emit DelegatedCeloRevoked(
+      delegator,
+      delegatee1,
+      FixidityLib.newFixedFraction(percentageToRevoke, 100).unwrap(),
+      amountToRevoke
+    );
+    lockedGold.revokeDelegatedGovernanceVotes(
+      delegatee1,
+      FixidityLib.newFixedFraction(percentageToRevoke, 100).unwrap()
+    );
+  }
+
+  function test_ShouldRevokeVotesCorrectlyWhenDelegateeVoting_WhenRevokingPercentageSuchAsThatWithNewlyLockedAmountIwWouldDecreaseBelowZero()
+    public
+  {
+    uint256 value = 1000;
+    uint256 percentageToRevoke = 2;
+    uint256 percentageToDelegate = 10;
+    uint256 delegatedAmount = (value * percentageToDelegate) / 100;
+    uint256 amountToRevoke = (value / 100) * percentageToRevoke;
+
+    uint256 percentageToRevokeAfterLock = 6;
+    uint256 votingAmount = (delegatedAmount * 2 - amountToRevoke);
+
+    uint256 amountFromDelegator1AfterRevoke = ((2 * delegatedAmount) / percentageToDelegate) *
+      (percentageToDelegate - percentageToRevoke - percentageToRevokeAfterLock);
+
+    vm.prank(delegator);
+    lockedGold.lock.value(value)();
+    vm.prank(delegator2);
+    lockedGold.lock.value(value)();
+
+    vm.prank(delegator);
+    lockedGold.delegateGovernanceVotes(
+      delegatee1,
+      FixidityLib.newFixedFraction(percentageToDelegate, 100).unwrap()
+    );
+    vm.prank(delegator2);
+    lockedGold.delegateGovernanceVotes(
+      delegatee1,
+      FixidityLib.newFixedFraction(percentageToDelegate, 100).unwrap()
+    );
+
+    vm.prank(delegator);
+    lockedGold.revokeDelegatedGovernanceVotes(
+      delegatee1,
+      FixidityLib.newFixedFraction(percentageToRevoke, 100).unwrap()
+    );
+
+    assertEq(
+      lockedGold.getAccountTotalGovernanceVotingPower(delegatee1),
+      delegatedAmount * 2 - amountToRevoke
+    );
+
+    vm.prank(delegator);
+    lockedGold.lock.value(value)();
+
+    governance.setTotalVotes(delegatee1, votingAmount);
+
+    vm.prank(delegator);
+    lockedGold.revokeDelegatedGovernanceVotes(
+      delegatee1,
+      FixidityLib.newFixedFraction(percentageToRevokeAfterLock, 100).unwrap()
+    );
+
+    assertEq(
+      lockedGold.getAccountTotalGovernanceVotingPower(delegatee1),
+      delegatedAmount + amountFromDelegator1AfterRevoke
+    );
+
+    assertDelegatorDelegateeAmounts(
+      delegator,
+      delegatee1,
+      percentageToDelegate - percentageToRevoke - percentageToRevokeAfterLock,
+      amountFromDelegator1AfterRevoke
+    );
+
+    assertDelegatorDelegateeAmounts(delegator2, delegatee1, percentageToDelegate, delegatedAmount);
+    assertEq(
+      lockedGold.totalDelegatedCelo(delegatee1),
+      delegatedAmount + amountFromDelegator1AfterRevoke
+    );
+    assertEq(
+      governance.removeVotesCalledFor(delegatee1),
+      delegatedAmount + amountFromDelegator1AfterRevoke
+    );
+
+  }
+
+  function test_ShouldRevokeVotesCorrectlyWhenDelegateeVoting_WhenRevokingPercentageSuchAsThatWithNewlyLockedAmountIwWouldNotDecreaseBelowZero()
```

Comment:
I'd create a new class for the `WhenRevokingPercentageSuchAsThatWithNewlyLockedAmountIwWouldNotDecreaseBelowZero`